# Scikit-Learn Jupyet Notebook
***

## Overview of the scikit-learn Python library

In [48]:
import pandas as pd

import numpy as np

In [49]:
from sklearn.datasets import load_wine

In [50]:
wine_data = load_wine()

In [51]:
dir(wine_data)

['DESCR', 'data', 'feature_names', 'frame', 'target', 'target_names']

In [52]:
print(wine_data.DESCR)

.. _wine_dataset:

Wine recognition dataset
------------------------

**Data Set Characteristics:**

    :Number of Instances: 178 (50 in each of three classes)
    :Number of Attributes: 13 numeric, predictive attributes and the class
    :Attribute Information:
 		- Alcohol
 		- Malic acid
 		- Ash
		- Alcalinity of ash  
 		- Magnesium
		- Total phenols
 		- Flavanoids
 		- Nonflavanoid phenols
 		- Proanthocyanins
		- Color intensity
 		- Hue
 		- OD280/OD315 of diluted wines
 		- Proline

    - class:
            - class_0
            - class_1
            - class_2
		
    :Summary Statistics:
    
    ============================= ==== ===== ======= =====
                                   Min   Max   Mean     SD
    ============================= ==== ===== ======= =====
    Alcohol:                      11.0  14.8    13.0   0.8
    Malic Acid:                   0.74  5.80    2.34  1.12
    Ash:                          1.36  3.23    2.36  0.27
    Alcalinity of Ash:            1

Key questions coming up
Why is there 50 in each category and 178 data points - empty, not valid, missing data - incoherent information like this - different classes have different amounts
<br>
the mean, max and min and sd do not align which tells us we haven't been handed a standard data set

In [53]:
inputs = wine_data.data # keep the more clear what we are dealing with

In [54]:
output = wine_data.target # outputs assigned

In [55]:
inputs.shape

(178, 13)

In [56]:
output.shape

(178,)

In [57]:
wine_data.feature_names

['alcohol',
 'malic_acid',
 'ash',
 'alcalinity_of_ash',
 'magnesium',
 'total_phenols',
 'flavanoids',
 'nonflavanoid_phenols',
 'proanthocyanins',
 'color_intensity',
 'hue',
 'od280/od315_of_diluted_wines',
 'proline']

In [58]:
df = pd.DataFrame(inputs, columns=wine_data.feature_names)

In [59]:
df = pd.concat([df, pd.DataFrame(output)], axis=1)

In [60]:
df

alcohol  malic_acid   ash  alcalinity_of_ash  magnesium  total_phenols  \
0      14.23        1.71  2.43               15.6      127.0           2.80   
1      13.20        1.78  2.14               11.2      100.0           2.65   
2      13.16        2.36  2.67               18.6      101.0           2.80   
3      14.37        1.95  2.50               16.8      113.0           3.85   
4      13.24        2.59  2.87               21.0      118.0           2.80   
5      14.20        1.76  2.45               15.2      112.0           3.27   
6      14.39        1.87  2.45               14.6       96.0           2.50   
7      14.06        2.15  2.61               17.6      121.0           2.60   
8      14.83        1.64  2.17               14.0       97.0           2.80   
9      13.86        1.35  2.27               16.0       98.0           2.98   
10     14.10        2.16  2.30               18.0      105.0           2.95   
11     14.12        1.48  2.32               16.8       95.0           2.20   
12     13.75        1.73  2.41               16.0       89.0           2.60   
13     14.75        1.73  2.39               11.4       91.0           3.10   
14     14.38        1.87  2.38               12.0      102.0           3.30   
15     13.63        1.81  2.70               17.2      112.0           2.85   
16     14.30        1.92  2.72               20.0      120.0           2.80   
17     13.83        1.57  2.62               20.0      115.0           2.95   
18     14.19        1.59  2.48               16.5      108.0           3.30   
19     13.64        3.10  2.56               15.2      116.0           2.70   
20     14.06        1.63  2.28               16.0      126.0           3.00   
21     12.93        3.80  2.65               18.6      102.0           2.41   
22     13.71        1.86  2.36               16.6      101.0           2.61   
23     12.85        1.60  2.52               17.8       95.0           2.48   
24     13.50        1.81  2.61               20.0       96.0           2.53   
25     13.05        2.05  3.22               25.0      124.0           2.63   
26     13.39        1.77  2.62               16.1       93.0           2.85   
27     13.30        1.72  2.14               17.0       94.0           2.40   
28     13.87        1.90  2.80               19.4      107.0           2.95   
29     14.02        1.68  2.21               16.0       96.0           2.65   
30     13.73        1.50  2.70               22.5      101.0           3.00   
31     13.58        1.66  2.36               19.1      106.0           2.86   
32     13.68        1.83  2.36               17.2      104.0           2.42   
33     13.76        1.53  2.70               19.5      132.0           2.95   
34     13.51        1.80  2.65               19.0      110.0           2.35   
35     13.48        1.81  2.41               20.5      100.0           2.70   
36     13.28        1.64  2.84               15.5      110.0           2.60   
37     13.05        1.65  2.55               18.0       98.0           2.45   
38     13.07        1.50  2.10               15.5       98.0           2.40   
39     14.22        3.99  2.51               13.2      128.0           3.00   
40     13.56        1.71  2.31               16.2      117.0           3.15   
41     13.41        3.84  2.12               18.8       90.0           2.45   
42     13.88        1.89  2.59               15.0      101.0           3.25   
43     13.24        3.98  2.29               17.5      103.0           2.64   
44     13.05        1.77  2.10               17.0      107.0           3.00   
45     14.21        4.04  2.44               18.9      111.0           2.85   
46     14.38        3.59  2.28               16.0      102.0           3.25   
47     13.90        1.68  2.12               16.0      101.0           3.10   
48     14.10        2.02  2.40               18.8      103.0           2.75   
49     13.94        1.73  2.27               17.4      

In [70]:
df.describe() # important in prediction of the class later on

sepal length (cm)  sepal width (cm)  petal length (cm)  \
count         150.000000        150.000000         150.000000   
mean            5.843333          3.057333           3.758000   
std             0.828066          0.435866           1.765298   
min             4.300000          2.000000           1.000000   
25%             5.100000          2.800000           1.600000   
50%             5.800000          3.000000           4.350000   
75%             6.400000          3.300000           5.100000   
max             7.900000          4.400000           6.900000   

       petal width (cm)      target  
count        150.000000  150.000000  
mean           1.199333    1.000000  
std            0.762238    0.819232  
min            0.100000    0.000000  
25%            0.300000    0.000000  
50%            1.300000    1.000000  
75%            1.800000    2.000000  
max            2.500000    2.000000

In [72]:
df.describe().style.format("{:.5f}")

In [61]:
pd.set_option('display.max_columns', None)

pd.set_option('display.max_rows', None)

In [62]:
from sklearn.datasets import load_iris

data = load_iris()

df = pd.DataFrame(data.data,columns=data.feature_names)

df['target'] = pd.Series(data.target)

In [63]:
print(df.head())

   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  
0       0  
1       0  
2       0  
3       0  
4       0  


In [64]:
def get_data(dataset):

    data = dataset

    df = pd.DataFrame(data.data,columns=data.feature_names)

    df['target'] = pd.Series(data.target)

    print(df.head())

In [65]:
get_data(load_iris())

   sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
0                5.1               3.5                1.4               0.2   
1                4.9               3.0                1.4               0.2   
2                4.7               3.2                1.3               0.2   
3                4.6               3.1                1.5               0.2   
4                5.0               3.6                1.4               0.2   

   target  
0       0  
1       0  
2       0  
3       0  
4       0  


In [66]:
from sklearn.datasets import load_iris, load_boston

get_data(load_boston())

      CRIM    ZN  INDUS  CHAS    NOX     RM   AGE     DIS  RAD    TAX  \
0  0.00632  18.0   2.31   0.0  0.538  6.575  65.2  4.0900  1.0  296.0   
1  0.02731   0.0   7.07   0.0  0.469  6.421  78.9  4.9671  2.0  242.0   
2  0.02729   0.0   7.07   0.0  0.469  7.185  61.1  4.9671  2.0  242.0   
3  0.03237   0.0   2.18   0.0  0.458  6.998  45.8  6.0622  3.0  222.0   
4  0.06905   0.0   2.18   0.0  0.458  7.147  54.2  6.0622  3.0  222.0   

   PTRATIO       B  LSTAT  target  
0     15.3  396.90   4.98    24.0  
1     17.8  396.90   9.14    21.6  
2     17.8  392.83   4.03    34.7  
3     18.7  394.63   2.94    33.4  
4     18.7  396.90   5.33    36.2  


Linear Regression

In [67]:
from sklearn.datasets import load_wine

In [68]:
wine_data = load_wine()

In [69]:
dir(wine_data)

['DESCR', 'data', 'feature_names', 'frame', 'target', 'target_names']

## References

# End

*******